core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/federation/SilentIteration.java

```diff
@@ -19,6 +22,8 @@
  */
 public class SilentIteration extends LookAheadIteration<BindingSet, QueryEvaluationException> {
```

Comment:
I only just noticed this after pushing an editorial fix, but apparently we have two `SilentIteration` classes:

- `org.eclipse.rdf4j.repository.sparql.federation.SilentIteration`
- `org.eclipse.rdf4j.query.algebra.evaluation.iterator.SilentIteration`

They look identical (barring the fixes you just did) - do we need both or can we get rid of the federated one?

Reply:
Ah - to answer my own question, this is probably a deliberate copy to avoid a cyclic dependency. 

Let's apply the fix to both copies in this PR. I'll look into a better solution so that we don't have to maintain separate copies separately.

Reply:
Looking a bit further, it seems the copy in `evaluation.iterator` is not actually used anywhere. Let's mark it deprecated, so we can decide to remove it in a later major release.